src/main/java/net/neoforged/moddevgradle/internal/utils/IdeDetection.java

```diff
@@ -27,4 +33,47 @@ public static boolean isIntelliJSync() {
     public static boolean isEclipse() {
         return System.getProperty("eclipse.application") != null;
     }
+
+    /**
+     * Try to find the IntelliJ project directory that belongs to this Gradle project.
+     * There are scenarios where this is impossible, since IntelliJ allows adding
+     * Gradle builds to IntelliJ projects in a completely different directory.
+     */
+    @Nullable
+    public static File getIntellijProjectDir(Project project) {
+        // Always try the root directory first, since it has the highest chance
+        var intellijProjectDir = getIntellijProjectDir(project.getRootDir());
+        if (intellijProjectDir != null) {
+            return intellijProjectDir;
+        }
+
+        // Try every included build
+        for (var includedBuild : project.getGradle().getIncludedBuilds()) {
+            if (!includedBuild.getProjectDir().equals(project.getRootDir())) {
+                intellijProjectDir = getIntellijProjectDir(includedBuild.getProjectDir());
+                if (intellijProjectDir != null) {
+                    return intellijProjectDir;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private static File getIntellijProjectDir(File gradleProjectDir) {
+        // Search the .idea folder belonging to this Gradle project
+        // In includeBuild scenarios, it might be above the project. It's also possible
+        // that it is completely separate.
+        var ideaDir = new File(gradleProjectDir, ".idea");
+        while (!ideaDir.exists()) {
+            gradleProjectDir = gradleProjectDir.getParentFile();
```

Comment:
Is this necessary for our test setup? If not, I'd rather not walk the disk up arbitrarily.